### ISSUE TITLE
KLayout PCells Issue

### ISSUE THREAD

**Issue by mole99** [IHP-GmbH/IHP-Open-PDK#290]
When I use the KLayout PCells, I get the following error:

```
ERROR: /home/leo/projects/IHP-Open-PDK/ihp-sg13g2/libs.tech/klayout/python/pypreprocessor/pypreprocessor/__init__.py:11: ModuleNotFoundError: No module named 'imp'
```

My Python version is 3.12.7 and it seems that `imp` has been deprecated in favor of `importlib` in version 3.12.

A quick solution is to change the import in pypreprocessors `__init__` to:

```
import importlib as imp
```

A better solution would be to replace all occurrences of `imp` with `importlib`.

There's a commit in your pypreprocessor fork that added the `imp` library, I'm not sure if this is for a reason: https://github.com/IHP-GmbH/pypreprocessor/commit/d6088b2b7c8a91ba159b6735b8a1cabcb5039d90

**Comment by akrinke** [comment]
Yes, I added the import because the file used `imp.lock_held()`. However, I got the same error a few days and removing `import imp` fixed it. Although I am not sure why, I think we can remove the import?

**Comment by akrinke** [comment]
I guess, the line `imp.lock_held()` wasn't executed in my case. So your suggestion might be better.

**Comment by mole99** [comment]
Good point! It seems that section isn't executed at all in our case. I can't really tell what its use is, you could probably try asking in the upstream repository, but it seems that it is no longer maintained as the last change was 6 years ago.

**Comment by sergeiandreyev** [comment]
@ThomasZecha, could you please take a look at this issue?

**Comment by adrienluitot** [comment]
Related to #187, might have to close both issues when the problem is fixed.

**Comment by ThomasZecha** [comment]
I'll will take a look.

**Comment by mole99** [comment]
Just checked and it works now as expected. Thanks!
